### ISSUE TITLE
Callback with new position/dimension data, before doing transition?

### ISSUE THREAD

**Issue by lunelson** [metafizzy/isotope#1001]
So I'm doing a layout of boxes with Isotope, each of which are also laid out internally with Isotope. The first overall layout is done by waiting on `$.Deferred`s from the internal layouts of each of the boxes, then laying out all the boxes in the page. Subsequently these boxes can change state on click (expand), causing them to re-layout internally, and then re-layout the page on the `layoutComplete` event from the box's layout. This has the effect that the box transitions its layout *and then* the page transitions; but what I'd really like would be if both the box's layout and the overall page layout (all boxes) could transition at the same time:  is there any possible way to do this? A way to somehow pass the re-calculated dimensions of the box to the overall page-layout, then let them both transition at once?

**Comment by lunelson** [comment]
test case here; currently with nested isotope transitions set to `0` 
http://codepen.io/lunelson/pen/PPxGZo?editors=101

**Comment by desandro** [comment]
You could hack in an event triggered after a `.layout` call.

Try adding this code:

``` js
var layout = Isotope.prototype.layout;
Isotope.prototype.layout = function() {
  layout.apply( this, arguments );
  this.dispatchEvent('didLayout')
};
```

Then you can listen for `didLayout` event, which is triggered immediately after setting new positions and new container size.

**Comment by lunelson** [comment]
Thanks for that @desandro it sounds like what I'm looking for; but if you check the new version, it only seems to fire the first time everything is laid out: subsequent changes don't trigger the event. I added a `console.log()` to your code so you can see it happen. It fires 25 + 1 times on first run but not after that:

http://codepen.io/lunelson/pen/KdbpVQ?editors=001

**Comment by desandro** [comment]
My bad. Try using `._layout`, with an underscore. I've updated the code above

**Comment by lunelson** [comment]
Oh great! I've updated my pen above and it seems solid—is there an option to control the easing of the transitions?

I'll ping back when I get a real project-implementation running, thanks for the response!

**Comment by desandro** [comment]
For easing, take a look at [`selectedAttraction` and `friction`](http://flickity.metafizzy.co/options.html#selectedattraction-friction)
